### ISSUE TITLE
iOS battery consumption

### ISSUE THREAD

**Issue by thejpster** [Automattic/pocket-casts-ios#663]
I played a podcast today and it chewed my battery. I've seen this previously as well, and noticed the phone get hot when playing.

3m on screen and 35m background and this app used 17% battery (according to the battery stats). That seems like a lot for a podcast?

iPhone SE 2022
iOS 16.2

**Comment by thejpster** [comment]
OK, so I don't fully understand the battery page on iOS 16.2. So I took some screenshots instead.

After I took these screenshots I locked the screen, which turned off the display. I did nothing else with the phone other than listen to the podcast. I received no messages, and the screen did not turn back on until I woke it up to take the next screenshot. I was playing through the loudspeaker.

There are four:

1. I start to play a podcast. It's on 77%.
2. I pause the podcast to have lunch. It has played for 10m09s and is on 74%, so the battery delta is between 2% and 4%.
3. After lunch, I resume the podcast. It's on 73%. It has idled for 15 minutes and the battery delta is between 0% and 2%. 
4. I pause the podcast again, to check the results. I have gone from one podcast to the next, so the total play time is 1h05m15s. It's dropped to 58%, so the battery delta is 14% to 16%.

The iPhone didn't get hot on this occasion, so maybe I didn't trigger the bug. Is 15% per hour expected in screen-off loudspeaker mode?

The general battery condition is reported at 95%, and I normally go from 7am to 11pm using about 50% of the battery, with some light use (a few hours of reading, but no podcasts or video), so I don't think a weak battery is the issue.

I'll try and get in the habit of screenshotting before and after when I listen to a podcast, to see if I can catch the issue.

**Comment by thejpster** [comment]
I listened to another 60m12s, but on headphones this time. This used just 4% of the battery.

**Comment by rviljoen** [comment]
Hey @thejpster - you did not attach the screenshots, so it is a bit difficult to follow along. I'll add a couple of comments which may help:
 - I never use my phone in loudspeaker for long periods, but 15% battery use in an hour does feel excessive.
 - >3m on screen and 35m background and this app used 17% battery (according to the battery stats). That seems like a lot for a podcast?"
 
 The battery stats does not show absolute value used (i.e. 17%), but the proportion of the battery used. I.e. if PocketCasts was using 17% battery during that hour (or the full day) then something else (or a combination of things) was using the other 83%.
- You can click on the individual activity bars on the battery usage screen to get the proportional usage per hour.
- I looked at an hour of background play on my phone and Pocket Casts used 56%, with the rest being system services and other background apps (Mail, push notifications, etc). The overall battery only went down by 3% for the hour though, listening via Bluetooth.
- The fact that the battery gets hot is an indication that something is in fact using a lot of energy, as opposed to the battery being old and draining quickly. Battery health of 95% is still good enough.
- Are you streaming the episodes or downloading ahead of time? If streaming, could it be that you are in poor WiFi or cell coverage while doing so causing the radios to work harder than needed?

**Comment by thejpster** [comment]
They were large so I omitted them, but if my notes were not sufficient I'll add them here.

I was on very good Wi-Fi the entire time. I believe it downloads when I first press play so it should be cached? Not sure.

I did suspect I was reading the battery page wrong. The later values are from the battery indicator in the top bar.

First, when I play.
![image](https://user-images.githubusercontent.com/959887/213993358-c2df0937-9cb3-4306-85f9-25e539d00112.png)

Second, when I pause. 
![image](https://user-images.githubusercontent.com/959887/213993371-09ef4e5d-8565-4eb3-bb0a-5fe864d0a1c4.png)

Third, when I resume after lunch. 
![image](https://user-images.githubusercontent.com/959887/213993389-b73388ca-99b3-475b-98ee-1c992f3230d4.png)

Fourth, when I stop.
![image](https://user-images.githubusercontent.com/959887/213993422-fb008538-c475-4c58-8c4a-e520b59ff8f2.png)

**Comment by rviljoen** [comment]
Does the battery stats show any other apps using a lot of battery during this time?

**Comment by thejpster** [comment]
Nothing. Here's the 12noon to 1pm time period corresponding with the screenshots above.

![image](https://user-images.githubusercontent.com/959887/213995060-acbd66fb-1502-4301-9766-f49c1c45a55e.png)

**Comment by thejpster** [comment]
Observations: the battery level is dropping fast on the green graph. The measure of 24 mins active corresponds with the amount of time I was actively playing a podcast during that hour (the 10m from before lunch, and 15m up to the top of the hour after lunch).

**Comment by thejpster** [comment]
The next hour is the same story.

![image](https://user-images.githubusercontent.com/959887/213995619-b0396597-b16c-43ec-89f0-ee17a1cccd65.png)

**Comment by rviljoen** [comment]
On first glance it does seem like there's an issue here. I think you should log a ticket with support so that they can review the debug logs from your device. When you log the ticket, please reference this issue number so that they can pick up the discussion and add here.

Profile->Settings->Help&Feedback->(Scroll to bottom)->Get in touch->Get Support

On this screen, you can also click on "Attached Logs" and scroll to "Debug Logs" to see if you can spot anything weird happening at that time.

**Comment by thejpster** [comment]
Done

**Comment by mdrockwell** [comment]
Another user reported this here: 6334420-zd-woothemes

> Hello, I listen to a lot of podcasts over the day but most of the time in background.
>
> But Pocket Casts is killing my battery, even in background. This is really annoying.

And

>  just use „trim silence“.
>
> I think this was getting more worse in the last weeks, it never happened so much before.
>
> Problem is that I’m using another Podcast App now and the Battery usage does not show the actual data for Pocket Casts right now.

Debug Logs: 777D96F9-7E05-4907-883A-6311F66646F2

**Comment by mdrockwell** [comment]
7105219-zd-a8c:

> since couple of months I’m constantly dealing with battery drain issues while using your app on my IPhone. Are there any plans to fixing that issue?

![image0](https://github.com/Automattic/pocket-casts-ios/assets/42454301/29a14c09-148b-43f0-a031-34eb7e46cf47)

They also shared battery capacity info — their battery is 81%.

**Comment by CameronMilliken** [comment]
>Hey! I have notices the last month or so that Pocket Casts are using a lot of my battery in the background which is getting really annoying.
> 
>My iPhone is reporting for today 4 minute on-screen and 2h34min background activity for this app and I think I have listened around 40 minutes. It’s similar numbers for the last 10 days
>
>Ios version: iOS 17.0.3 (21A360)
App version: 7,49

7208192-zd-a8c

**Comment by igotdes** [comment]
Also reported here in 8229269-zd-a8c:

> I notice that my iPhone 15 pro max always feels warm when using pocketcast. I notice even more since the last official is upgrade from Apple , 17.5 and 17.5.1.
> 
> My battery takes a hit when using your app and it is noticeable vs when I stream music from Apple Music (I download most episodes on pocketcast).

Here are the battery stats for the user:

<details>
<summary>Details</summary>

![image](https://github.com/Automattic/pocket-casts-ios/assets/74066241/c5b1ab82-b2e2-46fb-9fbf-3d767d506fc5)
![image](https://github.com/Automattic/pocket-casts-ios/assets/74066241/03e0d1f7-ed4a-4032-89a5-cc37f83e0132)

</details>

**Comment by scrutinizer80** [comment]
Same here. Pocketcasts does seem to consume quite a lot of battery during playback.

**Comment by scrutinizer80** [comment]
Same here. Pocketcasts does seem to consume quite a lot of power during playback.

**Comment by thejpster** [comment]
I've given up on pocket casts and am using the Apple Podcasts app instead. I've never noticed that eat my battery or make my phone hot.

**Comment by danielebogo** [comment]
@thejpster @scrutinizer80 did you use trim silence during the stream?

**Comment by scrutinizer80** [comment]
> @thejpster @scrutinizer80 did you use trim silence during the stream?

No. I usually play locally downloaded episodes while in airplane mode. So no streaming at all. Plain playback with no effects whatsoever.

**Comment by ashiagr** [comment]
9501412-zd-a8c

The ticket includes database export and logs for further investigation.
